packages/react/src/components/UIShell/SwitcherItem.tsx

```diff
@@ -0,0 +1,157 @@
+import React, { ForwardedRef, ElementType, WeakValidationMap } from 'react';
+import cx from 'classnames';
+import PropTypes from 'prop-types';
+import Link, { LinkProps } from './Link';
+import { usePrefix } from '../../internal/usePrefix';
+import { keys, match } from '../../internal/keyboard';
+
+export type SwitcherItemProps<E extends ElementType> = LinkProps<E> & {
+  /**
+   * Required props for accessibility label on the underlying item
+   */
+  'aria-label'?: string;
+  /**
+   * Required props for accessibility labelledby on the underlying item
+   */
+  'aria-labelledby'?: string;
+  /**
+   * Optionally provide a custom class to apply to the underlying `<li>` node
+   */
+  className?: string;
+  /**
+   * Specify the text content for the link
+   */
+  children: React.ReactNode;
+  /**
+   * Specify the tab index of the Link
+   */
+  tabIndex?: number;
+  /**
+   * Specify the index of the SwitcherItem
+   */
+  index?: number;
+  /**
+   * event handlers
+   */
+  handleSwitcherItemFocus?: (event: {
+    currentIndex: number;
+    direction: number;
+  }) => void;
+  /**
+   * event handlers
+   */
+  onKeyDown?: (event: KeyboardEvent) => void;
```

Comment:
Minor nit, but could these be alphabetized so they match the order presented in the proptypes?